mesonbuild/dependencies/misc.py

```diff
@@ -29,8 +29,8 @@ def netcdf_factory(env: 'Environment',
                    for_machine: 'mesonlib.MachineChoice',
                    kwargs: T.Dict[str, T.Any],
                    methods: T.List[DependencyMethods]) -> T.List['DependencyGenerator']:
-    language = kwargs.get('language', 'c')
-    if language not in ('c', 'cpp', 'fortran'):
+    language = kwargs.get('language')
+    if language not in (None, 'c', 'cpp', 'fortran'):
```

Comment:
This commit and the next commit use two different ways of dealing with the same issue. I prefer the approach of the next commit where `None` is explicitly translated to `'c'`. It may be that you don't do that here because because the value is sent down the call stack. However translating `None` to `'c'` here is a first small step to reduce the permeation of this weird convention.

Reply:
Sure, I've changed them both to the `if language is None: language = 'c'` approach